docker/migrate-to-minio.sh

```
@@ -0,0 +1,1211 @@
+    fi
+}
+
+# Function to check if MinIO is ready by probing the minio container
+wait_for_minio_ready() {
+    print_info "Waiting for MinIO to be ready..."
+    local max_attempts=30
+    local attempt=1
+
+    while [[ $attempt -le $max_attempts ]]; do
+        # Probe using mc from a one-off mc container to avoid relying on service state
+        if docker run --rm --network "$(basename $(pwd))_default" --entrypoint /bin/sh minio/mc:latest -lc \
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)